tests/end_to_end/test_op.py

```diff
@@ -97,3 +97,26 @@ def test_binops(execute):
     assert res.values["r10"] == 11
     assert res.values["r11"] == 9
     assert res.values["r12"] == 2
+
+
+def test_walrus(execute):
```

Comment:
There are two behaviors I wanted to tests hence added two code snippets. Should I make two separate tests for them?

Reply:
your code clearly checks the right hand and left hand expressions separately. having tests separated helps make debugging easier by having failing tests potentially pointing to the exact cause (eg (x:= 1) passes but (x:=y) fails so must be an issue with expression on RHS or something like that. It adds to the overhead when running them but we can parallelize them in the future.

Reply:
Ok, will make two separate tests!